### ISSUE TITLE
Downloading large files from google drive external storage causes internal server error

### ISSUE THREAD

**Issue by stefanriemens** [owncloud/core#8810]
Downloading a large file from google drive external storage results in error 500 Internal Server Error.
This is the error in /var/log/httpd/ssl_access_log:
PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 532690781 bytes) in /var/www/html/owncloud/apps/files_external/3rdparty/google-api-php-client/src/io/Google_CurlIO.php on line 112, referer: https://<redacted>/owncloud/index.php/apps/files?dir=/GoogleDrive/

I suppose this is caused by https://github.com/owncloud/core/blob/master/apps/files_external/lib/google.php#L393, which reads:
if (isset($downloadUrl)) {
$request = new \Google_HttpRequest($downloadUrl, 'GET', null, null);
$httpRequest = \Google_Client::$io->authenticatedRequest($request);
if ($httpRequest->getResponseHttpCode() == 200) {
$tmpFile = \OC_Helper::tmpFile($ext);
$data = $httpRequest->getResponseBody();
file_put_contents($tmpFile, $data);
return fopen($tmpFile, $mode);
}
}

In other words, it attempts to download the entire file contents into memory, and then write it out to a temp file, then to stream it the client. This seems a bit silly, and obviously won't work for large files (depending on the amount of memory php is allowed.)

I'm running owncloud 6.0.3, on php  5.4.

Thanks,
Stefan

**Comment by Xenopathic** [comment]
Closing in favour of #5949
